### PR TITLE
Hyperbolic

### PR DESCRIPTION
**Pull request by tobiasleibner**
Adds operators, evaluations, assembler specializations etc. to treat hyperbolic pdes. Not ready to be merged yet.

### CODE REVIEW

File: dune/gdt/discretefunction/default.hh [hunk 1]
```diff
@@ -34,7 +91,7 @@ class ConstDiscreteFunction
                                                 typename SpaceImp::DomainFieldType, SpaceImp::dimDomain,
                                                 typename SpaceImp::RangeFieldType, SpaceImp::dimRange, SpaceImp::dimRangeCols >
 {
-  static_assert(is_space< SpaceImp >::value, "SpaceImp has to be derived from SpaceInterface!");
+//  static_assert(is_space< SpaceImp >::value, "SpaceImp has to be derived from SpaceInterface!");
```
Comment (ftalbrecht) [inline dune/gdt/discretefunction/default.hh]: You could add `is_product_space` below `ProductInterface` (analogous to `is_space`) and try if
```c++
static_assert(is_space< ... >::value || is_product_space< ... >::value, "");
```
works.

File: dune/gdt/spaces/fv/defaultproduct.hh [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+
+template< class GridViewImp, class RangeFieldImp, size_t rangeDim, size_t rangeDimCols >
+class DefaultProductTraits
+    : public DefaultTraits< GridViewImp, RangeFieldImp, rangeDim, rangeDimCols >
+{
+  typedef DefaultTraits< GridViewImp, RangeFieldImp, rangeDim, rangeDimCols >  BaseType;
+public:
+  typedef DefaultProduct< GridViewImp, RangeFieldImp, rangeDim, rangeDimCols > derived_type;
+  using BaseType::polOrder;
```
Comment (ftalbrecht) [inline dune/gdt/spaces/fv/defaultproduct.hh]: You only have to pull in the definitions from the `BaseType` by using `using` if you want to use them locally (e.g., in a function in this class of to define another type). Everyone  accessing these types from the outside (e.g., `typename DefaultProductTraits< ... >::GridViewType` will be able to do so without you using `using`, the compiler just takes the definition from the `BaseType`. So you can probably drop a lot of these...

File: dune/gdt/localevaluation/laxfriedrichs.hh [hunk 3]
```diff
@@ -0,0 +1,387 @@
+// This file is part of the dune-gdt project:
+//   http://users.dune-project.org/projects/dune-gdt
+// Copyright holders: Felix Schindler
+// License: BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
+//
+// Contributors: Kirsten Weber, Tobias Leibner
```
Comment (ftalbrecht) [inline dune/gdt/localevaluation/laxfriedrichs.hh]: Is this copy/paste or did Kirsten really help you here? :)

**Comment by ftalbrecht** [comment]
Since you added your changes to pr #66, can this pr be deleted?

**Comment by tobiasleibner** [comment]
Yep.

**Pull request closed without merging**